src/Microsoft.AspNet.Mvc.Razor.Host/ViewHierarchyUtility.cs

```diff
@@ -77,18 +78,31 @@ private static IEnumerable<string> GetHierarchicalPath(string relativePath, stri
                 // If the specified path is for the file hierarchy being constructed, then the first file that applies
                 // to it is in a parent directory.
                 relativePath = Path.GetDirectoryName(relativePath);
+
                 if (string.IsNullOrEmpty(relativePath))
                 {
                     return Enumerable.Empty<string>();
                 }
             }
 
+            var builder = new StringBuilder(relativePath);
+            builder.Replace('\\', '/');
+
+            if (builder.Length > 0 && builder[0] != '/')
+            {
+                builder.Insert(0, '/');
+            }
```

Comment:
You may want to do this first to avoid all the blitting needed with an `Insert(0...)` - it's probably not significant though.  Is this a common case or uncommon?